src/main/java/org/mqttbee/api/mqtt/datatypes/MqttSharedTopicFilterBuilder.java

```diff
@@ -18,35 +18,53 @@
 package org.mqttbee.api.mqtt.datatypes;
 
 import org.mqttbee.annotations.NotNull;
+import org.mqttbee.annotations.Nullable;
+
+import java.util.function.Function;
 
 /**
  * @author Silvio Giebl
  */
-public class MqttSharedTopicFilterBuilder extends MqttTopicFilterBuilder {
+public class MqttSharedTopicFilterBuilder<P> extends MqttTopicFilterBuilder<P> {
+
+    private final Function<? super MqttSharedTopicFilter, P> parentConsumer;
```

Comment:
Because MqttTopicFilterBuilder overrides MqttSharedTopicFilterBuilder
so we need `Function<? super MqttSharedTopicFilter, P>`
but we inherit `Function<? super MqttTopicFilter, P>`

Reply:
Probably we can reuse the inherited version. Have a look here: https://github.com/mqtt-bee/mqtt-bee/tree/improvement/fluent-builders-use-parent-consumer-from-base-class. It might need some improvement.